src/process.rs

```diff
@@ -620,8 +620,35 @@ pub async fn kafka_logs(
 
     Ok(())
 }
-
 fn ungzip(compressed_data: Vec<u8>, key: String) -> Result<Vec<u8>, Error> {
```

Comment:
With current implementation this function never fails, so I would consider:
* making it return just `Vec<u8>` instead of a `Result`
* or making it actually fail, put the error messages in the returned error, and print them as warnings on callers side.